spring-cloud-zookeeper-discovery/src/main/java/org/springframework/cloud/zookeeper/discovery/configclient/ZookeeperConfigServerBootstrapper.java

```diff
@@ -67,54 +71,6 @@ public void initialize(BootstrapRegistry registry) {
 				ClassUtils.isPresent("org.springframework.cloud.bootstrap.marker.Marker", null)) {
 			return;
 		}
-		// create curator
-		CuratorFactory.registerCurator(registry, null, true, bootstrapContext -> isEnabled(bootstrapContext.get(Binder.class)));
```

Comment:
The `isEnabled` check in the code below always resulted in `false` because Boot had not fully loaded configuration properties at this point.  None of these registered beans actually returned anything.

Reply:
The isEnabled call would be executed as soon as the bean was needed.  In most cases that would have happened within `ZookeeperFunction.apply`.  However the `Binder` from the `BootstrapContext` in the `registerIfAbsent` function is not the same as the one that is used in the`Zookeeper.apply` method.  In the `apply` method we are getting the Binder while Boot is in the process of gathering all property sources.

See https://github.com/spring-cloud/spring-cloud-config/blob/main/spring-cloud-config-client/src/main/java/org/springframework/cloud/config/client/ConfigServerConfigDataLocationResolver.java#L240
and
https://github.com/spring-cloud/spring-cloud-config/blob/main/spring-cloud-config-client/src/main/java/org/springframework/cloud/config/client/ConfigServerConfigDataLocationResolver.java#L231

This `Binder` contains all resolved property sources at this point that may have not been applied to the BootstrapContext yet.  It is very confusing I know but this is the crux as to why I changed this code in the first place.

Reply:
>  Is it possible to make some changes there too, if that would lead to a more elegant solution here, or is that out of the question?

Possibly, but you would need to open an issue in Spring Boot and discuss it with them

> The other idea I had was that we could perhaps register another bean into the BootstrapContext, just before we register ConfigServerInstanceMonitor, something from where one could retrieve this "more knowledgeable" Binder. This could maybe be the ConfigDataLocationResolverContext resolverContext itself, maybe some new class' instance that is specific to Spring Cloud Config and this use-case.

I can try this an see if it would work, but it might take a few days